abcix/adapter/adapter.go

```diff
@@ -75,9 +88,34 @@ func (app *adaptedApp) CheckTx(req abcix.RequestCheckTx) (resp abcix.ResponseChe
 	return
 }
 
-func (app *adaptedApp) CreateBlock(req abcix.RequestCreateBlock, iter *abcix.MempoolIter) abcix.ResponseCreateBlock {
-	// TODO: defer to consensus engine for now
-	panic("implement me")
+func (app *adaptedApp) CreateBlock(
+	req abcix.RequestCreateBlock,
+	iter *abcix.MempoolIter,
+) (resp abcix.ResponseCreateBlock) {
+	if maxGas < 0 {
+		maxGas = 1<<(64-1) - 1
+	}
+	// Update remainBytes based on previous block
+	remainBytes := maxBytes -
+		maxOverheadForBlock -
+		maxHeaderBytes -
+		int64(len(req.LastCommitInfo.Votes))*maxVoteBytes -
+		int64(len(req.ByzantineValidators))*maxEvidenceBytes
+	remainGas := maxGas
+
+	for iter.HasNext() {
+		tx, err := iter.GetNextTransaction(remainBytes, remainGas)
+		if err != nil {
+			panic("failed to get next tx from mempool")
```

Comment:
add a TODO saying this in real world may never happen, and in the future we may need to handle this error more elegantly